Core 901 add portal routing handler

RoyEJohnson opened the pull request:
[CORE-901]

[CORE-901]: https://openstax.atlassian.net/browse/CORE-901?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

TomWoodward commented on src/app/components/shell/router.js:
```diff
@@ -133,39 +135,50 @@ function MainRoutes() {
-                <Route path="/textbooks/:title" element={<RedirectToCanonicalDetailsPage />} />
-                <Route path="/subjects/*" element={<ImportedPage name="subjects" />} />
-                <Route path="/k12/*" element={<ImportedPage name="k12" />} />
-                <Route path="/blog/*" element={<ImportedPage name="blog" />} />
```
i'm wondering if there are any of these, like k12, blog, general, that we explicitly don't want to work in the portal. i guess probably they're not hurting anything by being in there

RoyEJohnson commented:
I have changed the link handling: I created a `rewriteLinks` utility in the PortalContext that rewrites the href of any links in a given element to add the portal if portal is set.
The footer element calls it, and so does the raw-html component, which is used in the adoption and interest form headers.I directly rewrote links in a few other pages.
I updated DefaultLayout to use a landing page layout when the portal is set.
It assumes a flex page with a landing page layout is a portal, but also any path that has a portal and something that loads successfully will be treated as a portal.
So these are all portal pages:
http://localhost:3000/b2s-fall-24
http://localhost:3000/b2s-fall-24/details/books/biology-2e
http://localhost:3000/b2s-fall-24/careers
http://localhost:3000/not-a-page/careers

We might want to verify that the portal corresponds to an existing flex page before setting it?

TomWoodward reviewed (approved):
i do think we should do the thing when we're loading `/portal/page` where we load the page matching `portal` and use the layout configuration from that page when rendering the content from `page`

i also left another comment on the layout param loading that i think is not working correctly

i'm fine with merging this and making additional PRs for those issues if you would prefer though, this seems like it'll mostly work

RoyEJohnson commented on src/app/components/shell/router.js:
```diff
@@ -56,14 +58,14 @@ function useLoading(name) {
+    const {portal} = useParams();
     const {setLayoutParameters, layoutParameters} = useLayoutContext();
 
-    React.useEffect(
```
I'll see if I can address the flashing issue before I merge this.
